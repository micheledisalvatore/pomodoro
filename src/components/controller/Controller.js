import React from 'react';
import PropTypes from 'prop-types';

import { Button, PlayIcon, PauseIcon } from './Controller.style';

export const Controller = ({ isPaused, onClick }) => (
  <Button onClick={onClick}>
    { isPaused ? <PlayIcon /> : <PauseIcon /> }
  </Button>
);

Controller.propTypes = {
  isPaused: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
};

Controller.defaultProps = {
  isPaused: false,
};
