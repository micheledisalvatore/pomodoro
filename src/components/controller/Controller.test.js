import React from 'react';
import { shallow } from 'enzyme';

import { PlayIcon, PauseIcon } from './Controller.style';
import { Controller } from './Controller';

describe('Given a Controller component', () => {
  let component;
  let wrapper;
  const onClickMock = jest.fn();
  const requiredProps = {
    onClick: onClickMock,
  };

  describe('when it is rendered', () => {
    beforeEach(() => {
      component = <Controller {...requiredProps} />
      wrapper = shallow(component);
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })

    it('should render the pause icon', () => {
      expect(wrapper.find(PauseIcon)).toExist();
    })

    describe('and the user clicks on it', () => {
      beforeEach(() => {
        wrapper.simulate('click', 'foo');
      })

      afterEach(() => {
        onClickMock.mockReset();
      })

      it('should invoke onClickMock', () => {
        expect(onClickMock).toBeCalledWith('foo');
      })
    })

    describe('and the component is in pause', () => {
      beforeEach(() => {
        wrapper.setProps({
          isPaused: true,
        });
      });

      it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
      })

      it('should render the play icon', () => {
        expect(wrapper.find(PlayIcon)).toExist();
      })
    })
  })
})
