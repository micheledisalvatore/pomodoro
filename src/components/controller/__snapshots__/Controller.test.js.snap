// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Controller component when it is rendered and the component is in pause should match the snapshot 1`] = `
<styled.button
  onClick={[MockFunction]}
>
  <PlayIcon />
</styled.button>
`;

exports[`Given a Controller component when it is rendered should match the snapshot 1`] = `
<styled.button
  onClick={[MockFunction]}
>
  <PauseIcon />
</styled.button>
`;
